### PR TITLE
Fix/dev erg

### PR DESCRIPTION
**Pull request by nick-stebbings**
- Based off @pospi 's pnpm changes and made further changes to build scripts
 -- Now there is an `npm run dev` which runs in the browser instead of Tauri, but I need help before merge to fix an issue with cloned cells getting the correct websocket URL (@weswalla @adaburrows )
 -- `npm run start` still loads up two Tauri windows
-- `npm run test:ui` now runs frontend tests, (some of which are failing due to a refactor of the dashboard code into two components before Dweb. See below for new CI practices

- Made updates to scoped element components to use the newer version  (coming from lit, not open-wc)
- I also updated `flake.nix` using one from the holochain scaffolding CLI. This followed up from a conversation where somebody was commenting on the 50GB of nix/store that needed to be downloaded when using several of our repos, and it would make sense to change all flake.nix to this one IMO.

Continuous Integration:
- Added husky: using a git pre-commit hook we can now stop commits that will fail any frontend UI tests.
- All tests are now passing, and from now on that will always be the case (or it won't let you commit! This can also be changed to e.g. only run the test suite on merging into develop)

To be sorted before merge:
- [x] [ui/app/src/elements/components/table-filter-map.ts](https://github.com/neighbour-hoods/nh-launcher/compare/develop...neighbour-hoods:nh-launcher:fix/dev-erg#diff-32fdeced54f88f9f5c03f26fb3e7ae6a53362ee818329c6183a259da9b4919b9) needs fixing 
- [x] launcher running in browser (mentioned above) needs correct websocket URL for cloned cells
- [x] use environment variables for admin and app websocket urls

### CODE REVIEW

**Comment by weswalla** [comment]
@nick-stebbings is the websocket URL issue occurring when creating a new group?
> -- Now there is an npm run dev which runs in the browser instead of Tauri, but I need help before merge to fix an issue with cloned cells getting the correct websocket URL

File: package.json [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-    "package": "npm run build:happ && npm run package -w we && hc web-app pack workdir",
+    "dev": "AGENTS=2 npm run network:dev",
+    "network": "hc s clean && npm run build:happ && npm run build -w @neighbourhoods/nh-launcher-applet && npm run build -w nh-launcher && concurrently \"npm run launch:happ\" \"npm run build:watch -w ui/app\"",
+    "network:dev": "hc s clean && npm run build:happ && npm run build -w @neighbourhoods/nh-launcher-applet && UI_PORT=8888 concurrently \"open-cli http://localhost:8888 && cd ui/app && npm start\" \"ADMIN_PORT=9000 HC_PORT=9001 npm run start:dev\"",
+    "launch:happ": "RUST_LOG=warn WASM_LOG=debug echo \"pass\" | hc launch --piped -n $AGENTS ./workdir/we.happ --ui-path ./ui/app/dist network mdns",
+    "launch:dev": "RUST_LOG=warn WASM_LOG=debug echo \"pass\" | hc launch --piped -n $AGENTS ./workdir/we.happ --ui-port $UI_PORT network mdns",
+    "start:dev": "RUST_LOG=warn WASM_LOG=debug echo \"pass\" | hc s --piped -f=$ADMIN_PORT generate ./workdir/we.happ --run $HC_PORT -a we network mdns",
```
Comment (weswalla) [inline package.json:15]: this is great. Thanks for moving us back to the browser for development!

**Comment by weswalla** [comment]
overall looks good. I'll take a crack at the websocket issue in the browser after tomorrow's ecosystem session. Liking the direction you are taking this in though, it's going to make developing so much more enjoyable!

**Comment by weswalla** [comment]
@nick-stebbings regarding the error, how certain are you that it is a websocket issue? I'm asking because I know that when we use tauri through the `hc launch` cli it handles the zome signature authorization step for us, but otherwise we will have to do that manually I believe.

**Comment by nick-stebbings** [comment]
> @nick-stebbings is the websocket URL issue occurring when creating a new group?
> 
> > -- Now there is an npm run dev which runs in the browser instead of Tauri, but I need help before merge to fix an issue with cloned cells getting the correct websocket URL

Yes that is where I have encountered it and it is indeed the  zome signature authorization step, as it explicitly says that in the error. So how do we go about doing that manually?

**Comment by weswalla** [comment]
Its just an `adminWebsocket` call: https://github.com/holochain/holochain-client-js/blob/main/docs/client.adminwebsocket.authorizesigningcredentials.md
example in acorn: https://github.com/lightningrodlabs/acorn/blob/ebf2b2a051144ecd8161d5d6e55882a05149421b/web/src/index.tsx#L73C34-L73C34
I'll take a look today.

**Comment by weswalla** [comment]
@nick-stebbings got the zome signing situation all sorted, except for applet cells, which will have to be either added to the applet code or a change to the applet interface to allow nh-launcher to manage the authorization for newly added applet cells.

**Comment by nick-stebbings** [comment]
Great work! This already allows me to style the dashboard from the browser which is a massive win :)

**Pull request merged**